pipeline/bicleaner/bicleaner.sh

```diff
@@ -44,10 +44,32 @@ else
     tcol=1
   fi
 
+  #Export cuda visible devices if not set
+  if [ ${#CUDA_VISIBLE_DEVICES} == 0 ]; then   export CUDA_VISIBLE_DEVICES=$(nvidia-smi --query-gpu=index --format=csv,noheader); fi
+
   echo "### Classifying"
-  paste <(pigz -dc "${corpus_prefix}.${SRC}.gz") <(pigz -dc "${corpus_prefix}.${TRG}.gz") |
-    ${cmd} --scol ${scol} --tcol ${tcol} --processes "${threads}"  - - "${pack_dir}"/*.yaml |
-    pigz >"${output_prefix}.scored.gz"
+  if [[ "${type}" == 'bicleaner-ai' && ${#CUDA_VISIBLE_DEVICES} > 1 ]]; then # Use gnu-parallel'd bicleaner-ai if we have more than 1 GPU
+       #Convert CUDA_VISIBLE_DEVICES to an array
+       export CUDA_VISIBLE_ARRAY=($CUDA_VISIBLE_DEVICES)
+       #Turn on tensorflow logging in bicleaner-ai
+       export TF_CPP_MIN_LOG_LEVEL=0
+       #This function expects a bicleaner yaml and a 1-based index into the CUDA_VISIBLE_ARRAY
+       #Example: /mnt/nanna0/nbogoych/data/data/fr-en/fr-en-prod/biclean/pack/metadata.yaml index_in_CUDA_VISIBLE_ARRAY+1
+       biclean() {
+               export CUDA_VISIBLE_ARRAY=($CUDA_VISIBLE_DEVICES)
+               export CUDA_VISIBLE_DEVICES=${CUDA_VISIBLE_ARRAY[$(($2-1))]}
+               bicleaner-ai-classify --scol 2 --tcol 1 - - $1
```

Comment:
should be `--scol ${scol} --tcol ${tcol}`

Reply:
Fixed.